components/style/properties/properties.mako.rs

```diff
@@ -206,31 +206,72 @@ pub mod animated_properties {
 <%
     from itertools import groupby
 
+    copyTypes = set([
+        "AlignContent",
+        "AlignItems",
+        "AlignSelf",
+        "BackgroundRepeat",
+        "BorderImageRepeat",
+        "BorderStyle",
+        "Contain",
+        "FontStyleAdjust",
+        "FontSynthesis",
+        "FontWeight",
+        "GridAutoFlow",
+        "ImageOrientation",
+        "InitialLetter",
+        "Integer",
+        "IntegerOrAuto",
+        "JustifyContent",
+        "JustifyItems",
+        "JustifySelf",
+        "MozForceBrokenImageIcon",
+        "MozScriptLevel",
+        "MozScriptMinSize",
+        "MozScriptSizeMultiplier",
+        "NonNegativeNumber",
+        "Opacity",
+        "OutlineStyle",
+        "OverscrollBehavior",
+        "Percentage",
+        "PositiveIntegerOrAuto",
+        "SVGPaintOrder",
+        "ScrollSnapType",
+        "TextDecorationLine",
+        "TouchAction",
+        "TransformStyle",
+        "XSpan",
+        "XTextZoom",
+    ])
+
     variants = []
     for property in data.longhands:
         if property.predefined_type and not property.is_vector:
+            copy = property.predefined_type in copyTypes
             ty = "::values::specified::{}".format(property.predefined_type)
         else:
+            copy = bool(property.keyword) and not property.is_vector
             ty = "longhands::{}::SpecifiedValue".format(property.ident)
         if property.boxed:
             ty = "Box<{}>".format(ty)
         variants.append({
             "name": property.camel_case,
             "type": ty,
             "doc": "`" + property.name + "`",
+            "copy": copy,
         })
 
     groups = {}
```

Comment:
It'd be nice to document this whole algorithm, or just the intended result, fwiw :)

IIUC the idea es that the copy variants end up sorted first, then the rest sorted by common type.